### PR TITLE
Minor enhancements to bam_cat

### PR DESCRIPTION
**Pull request by jenniferliddle**
Now supports --output-fmt-option and --output-fmt command line options.
Added @PG line to output file

### CODE REVIEW

File: bam_cat.c [hunk 1]
```diff
@@ -604,13 +648,13 @@ int main_cat(int argc, char *argv[])
     switch (hts_get_format(in)->format) {
```
Comment (valeriuo) [inline bam_cat.c]: The output file format is still decided by the format of the first input file, which makes the `--output-fmt` option useless.

Comment (jenniferliddle) [reply bam_cat.c]: Done

File: bam_cat.c [hunk 2]
```diff
@@ -313,6 +342,7 @@ int cram_cat(int nfn, char * const *fn, const bam_hdr_t *h, const char* outcram)
     cram_set_option(out_c, CRAM_OPT_VERSION, vers);
     //fprintf(stderr, "Creating cram vers %s\n", vers);
 
+    if (addNewPG(new_h, arg_list)) return -1;
```
Comment (valeriuo) [inline bam_cat.c]: You should switch to using the new header API method `bam_hdr_add_pg`.

**Comment by valeriuo** [comment]
Merged as 3e81cc099fb4d485019f3a73d66139d74ec08d05 into https://github.com/samtools/samtools/pull/998

**Pull request closed without merging**